web/src/unread.ts

```diff
@@ -509,20 +570,28 @@ class UnreadTopicCounter {
         return streams_with_mentions;
     }
 
-    get_streams_with_unmuted_mentions() {
+    get_streams_with_unmuted_mentions(): Set<number> {
         // Collect the set of streams containing at least one mention
         // that is not in a muted topic or non-unmuted topic in a
         // muted stream.
-        const streams_with_unmuted_mentions = new Set();
+        const streams_with_unmuted_mentions: Set<number> = new Set();
         for (const message_id of unread_mentions_counter) {
             const stream_id = this.bucketer.reverse_lookup.get(message_id);
             if (stream_id === undefined) {
                 // This is a direct message containing a mention.
                 continue;
             }
 
-            const stream_bucketer = this.bucketer.get_bucket(stream_id);
+            const stream_bucketer = this.bucketer.get_bucketer(stream_id);
+            if (!stream_bucketer) {
+                // TODO(evy): warn here?
+                continue;
+            }
             const topic = stream_bucketer.reverse_lookup.get(message_id);
+            if (!topic) {
+                // TODO(evy): warn here?
+                continue;
+            }
```

Comment:
These can probably both be asserts; we believed that this algorithm should in its construction have those items be in the bucketers. That'll avoid creating lines that test coverage wants to think about for cases that should be impossible.

Reply:
I guess there is a question for whether we want an assert here or to offer a `.get(message_id, assert_exists=true)` sort of API.

Reply:
An `{assert_exists: true}` flag is very bad from a static typing perspective because it dynamically changes whether the caller needs to deal with an `undefined` return. (You can express this inconveniently with overloads, but even then, TypeScript doesn’t fully check that the function body really satisfies the overloads.)

Reply:
So @andersk would you recommend handling it here instead? Do you think a warn would make sense, or just `continue` as is written?